### PR TITLE
MGMT-4049 & OCPBUGSM-22823 - Use ID to identify disks instead of disk path

### PR DESCRIPTION
**Pull request by michaellevy101**

### CODE REVIEW

**Comment by omertuc** [comment]
Commit message says `The id field identify the disk to the UI instead of computing it from the by-path and by-id field`

This is not accurate, the id field is also used internally by us to determine which disk we should install on. This is not just for UI indication.

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -2103,60 +2094,65 @@ func (b *bareMetalInventory) updateHostRoles(ctx context.Context, params install
...
-				params.ClusterUpdateParams.HostsNames[i].Hostname, params.ClusterUpdateParams.HostsNames[i].ID,
+				hostName.Hostname, hostName.ID,
 				params.ClusterID)
 			return common.NewApiError(http.StatusConflict, err)
 		}
 	}
 	return nil
 }
 
+func getHost(hostId string, clusterId string, db *gorm.DB, log logrus.FieldLogger) (*models.Host, error){
```
Comment (omertuc) [inline internal/bminventory/inventory.go:2118]: @YuviGold didn't you already create a similar utility function? Or was that just for clusters?

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -1696,15 +1696,11 @@ func (b *bareMetalInventory) refreshClusterHosts(ctx context.Context, cluster *c
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
 	for _, h := range cluster.Hosts {
-		var host models.Host
-		var err error
-		if err = tx.Take(&host, "id = ? and cluster_id = ?",
-			h.ID.String(), cluster.ID.String()).Error; err != nil {
-			log.WithError(err).Errorf("failed to find host <%s> in cluster <%s>",
-				h.ID.String(), cluster.ID.String())
+		var host, err = getHost(h.ID.String(), cluster.ID.String(), tx, log)
```
Comment (omertuc) [inline internal/bminventory/inventory.go:1699]: Is this code change required for this PR? If possible please isolate it to its own commit with its own message explaining why it changed. 

If it's not required, please add it to a separate PR

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -2078,22 +2074,17 @@ func optionalParam(data *string, field string, updates map[string]interface{}) {
...
-
 	for i := range params.ClusterUpdateParams.HostsRoles {
-		log.Infof("Update host %s to role: %s", params.ClusterUpdateParams.HostsRoles[i].ID,
-			params.ClusterUpdateParams.HostsRoles[i].Role)
-		var host models.Host
-		err := db.First(&host, "id = ? and cluster_id = ?",
-			params.ClusterUpdateParams.HostsRoles[i].ID, params.ClusterID).Error
+		hostRole := params.ClusterUpdateParams.HostsRoles[i]
+		log.Infof("Update host %s to role: %s", hostRole.ID, hostRole.Role)
+		var host, err = getHost(hostRole.ID.String(), params.ClusterID.String(), db, log)
```
Comment (omertuc) [inline internal/bminventory/inventory.go:2080]: Same for this change - it's a nice refactor but doesn't seem relevant to the PR / commit message

File: internal/bminventory/inventory.go [hunk 4]
```diff
@@ -2078,22 +2074,17 @@ func optionalParam(data *string, field string, updates map[string]interface{}) {
...
 
 func (b *bareMetalInventory) updateHostRoles(ctx context.Context, params installer.UpdateClusterParams, db *gorm.DB, log logrus.FieldLogger) error {
-
 	for i := range params.ClusterUpdateParams.HostsRoles {
-		log.Infof("Update host %s to role: %s", params.ClusterUpdateParams.HostsRoles[i].ID,
-			params.ClusterUpdateParams.HostsRoles[i].Role)
-		var host models.Host
-		err := db.First(&host, "id = ? and cluster_id = ?",
-			params.ClusterUpdateParams.HostsRoles[i].ID, params.ClusterID).Error
+		hostRole := params.ClusterUpdateParams.HostsRoles[i]
```
Comment (omertuc) [inline internal/bminventory/inventory.go:2078]: Same for this change - it's a nice refactor but doesn't seem relevant to the PR / commit message

File: internal/bminventory/inventory.go [hunk 5]
```diff
@@ -2103,60 +2094,65 @@ func (b *bareMetalInventory) updateHostRoles(ctx context.Context, params install
...
-				}
-				installationDiskPath = hostutil.FindDiskPathByID(*diskConfig.ID, &host)
-			}
+		disksToInstallOn := funk.Find(disksConfig.DisksConfig, func(diskConfigParams *models.DiskConfigParams) bool {
+			return models.DiskRoleInstall == diskConfigParams.Role
+		}).([]*models.DiskConfigParams)
+
+		installationDiskPath := ""
+
+		if len(disksToInstallOn) > 1{
```
Comment (omertuc) [inline internal/bminventory/inventory.go:2144]: file doesn't look like it's been through gofmt `1{`

File: internal/bminventory/inventory.go [hunk 6]
```diff
@@ -2103,60 +2094,65 @@ func (b *bareMetalInventory) updateHostRoles(ctx context.Context, params install
...
+			return common.NewApiError(http.StatusConflict, fmt.Errorf("duplicate setting of installation path by the user"))
```
Comment (omertuc) [inline internal/bminventory/inventory.go:2145]: `errorf` without formatting

File: internal/bminventory/inventory.go [hunk 7]
```diff
@@ -2103,60 +2094,65 @@ func (b *bareMetalInventory) updateHostRoles(ctx context.Context, params install
...
+		} else if len(disksToInstallOn) == 1 {
+			installationDiskPath = *disksToInstallOn[0].ID
 		}
 
-		log.Infof("Update host %s to install from path %s",
-			params.ClusterUpdateParams.DisksSelectedConfig[i].ID, installationDiskPath)
-		err = b.hostApi.UpdateInstallationDiskPath(ctx, db, &host, installationDiskPath)
+		log.Infof("Update host %s to install from path %s", hostId, installationDiskPath)
+		err = b.hostApi.UpdateInstallationDiskPath(ctx, db, host, installationDiskPath)
```
Comment (omertuc) [inline internal/bminventory/inventory.go:2151]: Please modify the description for the installationDiskPath field of the host to explain that its name is misleading and was kept for backwards compatibility purposes, and what it should be called today is "installationDiskID"

**Comment by omertuc** [comment]
Please edit PR description to include the commit messages

File: internal/bminventory/inventory.go [hunk 8]
```diff
@@ -2103,60 +2094,65 @@ func (b *bareMetalInventory) updateHostRoles(ctx context.Context, params install
...
 		if err != nil {
 			log.WithError(err).Errorf("failed to set installation disk path <%s> host <%s> in cluster <%s>",
```
Comment (omertuc) [inline internal/bminventory/inventory.go:2153]: `disk id` rather than `disk path`

File: internal/bminventory/inventory.go (hunk 5 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go:2144]: done

File: internal/bminventory/inventory.go (hunk 6 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go:2145]: done

File: internal/host/host.go [hunk 9]
```diff
@@ -255,19 +255,19 @@ func (m *Manager) populateDisksEligibility(inventoryString string) (string, erro
 // was set is no longer part of the inventory, the new installation disk is re-evaluated because it is not longer
 // a valid choice.
 func determineDefaultInstallationDisk(previousInstallationDisk string, validDisks []*models.Disk) string {
```
Comment (omertuc) [inline internal/host/host.go:257]: We need to think carefully about backwards compatibility concerns. I don't think it's possible to simply change the meaning of this field all of a sudden. It has the potential to break clusters that are discovering during the rollout of these changes.

I think we might need to add a new field here - installation disk ID, that will live along side installation disk path, which we will have to more carefully deprecate...

File: internal/bminventory/inventory.go (hunk 7 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go:2151]: I documented the change in the swagger.yaml file. change the method name to UpdateInstallationDiskId? (so only the field name will remain the same)

Comment (omertuc) [inline internal/bminventory/inventory.go:2151]: Can we rename this function? DiskID

Comment (omertuc) [reply internal/bminventory/inventory.go:2151]: > I documented the change in the swagger.yaml file.
I don't see it in the PR diff. I see that you documented the ID field in the inventory disk object, but not in the host "installationDiskPath" field

Comment (omertuc) [reply internal/bminventory/inventory.go:2151]: > Change the method name to UpdateInstallationDiskId? (so only the field name will remain the same)
Yeah see my comment below

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go:1699]: done

File: internal/bminventory/inventory.go (hunk 4 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go:2078]: done

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (YuviGold) [reply internal/bminventory/inventory.go]: Right. Please use `common.GetHostFromDB`
https://github.com/openshift/assisted-service/blob/927f8bbdf395e2ee830749eb672283d1089396b8/internal/common/db.go#L135-L142

File: internal/host/hostutil/host_utils.go [hunk 10]
```diff
@@ -68,6 +68,19 @@ func ValidateHostname(hostname string) error {
...
 
+func GetInventoryDisk(host *models.Host, diskId string) (*models.Disk, error) {
+	var inventory models.Inventory
+	err := json.Unmarshal([]byte(host.Inventory), &inventory)
+	if err != nil {
+		return nil, err
+	}
+
+	result := funk.Find(inventory.Disks, func(disk models.Disk) bool {
+		return disk.ID == diskId
```
Comment (omertuc) [inline internal/host/hostutil/host_utils.go:79]: As part of the backwards compatibility concerns I raised above, this function should probably do something like:
If the ID didn't correspond to any disk, it should try comparing it to disk.Path. This way, hosts that had their installation path determined before the new changed rolled out, would still find a disk with this function.

File: internal/host/transition.go [hunk 11]
```diff
@@ -112,38 +112,35 @@ func (th *transitionHandler) PostRegisterDuringReboot(sw stateswitch.StateSwitch
...
 
-	statusInfo := fmt.Sprintf("Expected the host to boot from disk, but it booted the installation image - please reboot and fix boot order to boot from disk %s",
-		sHost.host.InstallationDiskPath)
-
-	if installationDisk != nil && installationDisk.Serial != "" {
-		statusInfo += fmt.Sprintf(" (%s)", installationDisk.Serial)
+	var statusInfo strings.Builder;
+	statusInfo.WriteString("Expected the host to boot from disk, but it booted the installation image - please reboot and fix boot order to boot from disk")
+	if installationDisk.Model != "" {
+		statusInfo.WriteString(installationDisk.Model)
```
Comment (omertuc) [inline internal/host/transition.go:131]: missing space before this... I think?

What is the error even supposed to look like?

File: internal/host/transition.go [hunk 12]
```diff
@@ -112,38 +112,35 @@ func (th *transitionHandler) PostRegisterDuringReboot(sw stateswitch.StateSwitch
...
+	}
+	if installationDisk.Serial != "" {
+		statusInfo.WriteString(" ")
```
Comment (omertuc) [inline internal/host/transition.go:134]: This space might be unnecessary?

If you're looking for multiple strings, space separated, with some of them potentially missing, I'd suggest just initializing an empty string array, appending to it, then using `strings.Join(' ', myStrings)`

File: internal/host/host_test.go [hunk 13]
```diff
@@ -996,33 +998,33 @@ var _ = Describe("UpdateInventory", func() {
 				expectedInstallationDisk: ""},
 			{testName: "No previous installation disk, one disk in inventory",
 				currentInstallationDisk:  "",
-				inventoryDisks:           []*models.Disk{{Name: diskName}},
-				expectedInstallationDisk: hostutil.GetDeviceFullName(diskName)},
+				inventoryDisks:           []*models.Disk{{ID: diskId, Name: diskName}},
```
Comment (omertuc) [inline internal/host/host_test.go:1001]: I think the `diskName` can be removed here, it's irrelevant for the test

File: internal/host/host_test.go [hunk 14]
```diff
@@ -996,33 +998,33 @@ var _ = Describe("UpdateInventory", func() {
...
+				currentInstallationDisk:  diskId,
+				inventoryDisks:           []*models.Disk{{ID: diskId, Name: diskName}, {Name: otherDiskName}},
+				expectedInstallationDisk: diskId},
 			{testName: `Previous installation disk is set, new inventory still contains that disk, but there's another
 						disk with higher priority`,
-				currentInstallationDisk:  hostutil.GetDeviceFullName(diskName),
-				inventoryDisks:           []*models.Disk{{Name: otherDiskName}, {Name: diskName}},
-				expectedInstallationDisk: hostutil.GetDeviceFullName(diskName)},
+				currentInstallationDisk:  diskId,
+				inventoryDisks:           []*models.Disk{{Name: otherDiskName}, {ID: diskId, Name: diskName}},
```
Comment (omertuc) [inline internal/host/host_test.go:1018]: forgot otherDiskID here and in some other places too

File: internal/host/hostcommands/install_cmd.go [hunk 15]
```diff
@@ -248,14 +248,13 @@ func (i *installCmd) getDiskUnbootableCmd(ctx context.Context, host models.Host)
 	for _, disk := range inventory.Disks {
 		isFcIscsi := strings.Contains(disk.ByPath, "-fc-") || strings.Contains(disk.ByPath, "-iscsi-")
 		if disk.Bootable && !isFcIscsi && !disk.IsInstallationMedia {
-			dev := hostutil.GetDeviceFullName(disk.Name)
-			formatCmds += fmt.Sprintf("dd if=/dev/zero of=%s bs=512 count=1 ; ", dev)
+			formatCmds += fmt.Sprintf("dd if=/dev/zero of=%s bs=512 count=1 ; ", disk.ID)
```
Comment (omertuc) [inline internal/host/hostcommands/install_cmd.go:251]: Why `disk.ID`? this should be `disk.Path`

File: internal/host/hostcommands/install_cmd.go [hunk 16]
```diff
@@ -248,14 +248,13 @@ func (i *installCmd) getDiskUnbootableCmd(ctx context.Context, host models.Host)
...
 			i.eventsHandler.AddEvent(
 				ctx,
 				host.ClusterID,
 				host.ID,
 				models.EventSeverityInfo,
-				fmt.Sprintf("Performing quick format of disk %s on host %s", dev, hostutil.GetHostnameForMsg(&host)),
+				fmt.Sprintf("%s: Performing quick format of disk %s", hostutil.GetHostnameForMsg(&host), disk.ID),
```
Comment (omertuc) [inline internal/host/hostcommands/install_cmd.go:257]: Also here. The user doesn't care about disk ID's, they only care about paths.

**Comment by rollandf** [comment]
Make sure to check also the Agent CRD and agent_controller code and tests.
The installation disk is also supported in Kube APIs

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go]: It's a different Host type: common.Host instead of models.Host

Comment (omertuc) [reply internal/bminventory/inventory.go]: > It's a different Host type: common.Host instead of models.Host

which is just a models.Host with an extra field... @rollandf what is this weird distinction between `common.Host` and `models.Host`? why isn't the Approved field in the `swagger.yaml`?

File: internal/host/host.go (hunk 9 above)
Comment (slaviered) [reply internal/host/host.go:257]: Another option would be to have intelligent migration to the installation_disk_path to reflect the new calculated value.  You are bound to have the same problem with installation_disk_id with discovered hosts. If they were selected before than suddenly the selection disappear with the migration.

File: internal/host/hostutil/host_utils.go (hunk 10 above)
Comment (slaviered) [reply internal/host/hostutil/host_utils.go:79]: Agreed. But having migration script to change the content to installation_disk_path can resolve this issue without need for convoluted code in runtime

Comment (slaviered) [reply internal/host/hostutil/host_utils.go:79]: Or go with the extra new field. Each of the options is better than changing the code here

File: internal/bminventory/inventory.go (hunk 8 above)
Comment (slaviered) [reply internal/bminventory/inventory.go:2153]: If the Find function did not found a configuration disk with role==install it should not set the installation_path to "" but do nothing

File: internal/host/host_test.go (hunk 14 above)
Comment (slaviered) [reply internal/host/host_test.go:1018]: +1

File: internal/host/hostcommands/install_cmd.go (hunk 15 above)
Comment (slaviered) [reply internal/host/hostcommands/install_cmd.go:251]: Add a test where the installation_disk_path is set to path and see how do you handle the conversion. This will also test the upgrade scenario

File: internal/host/transition.go (hunk 11 above)
Comment (slaviered) [reply internal/host/transition.go:131]: Instead of the model or serial which does not mean anything to the user I would add here the disk path (since now the installation_disk_path is actually an ID and it is not human readable

File: swagger.yaml [hunk 17]
```diff
@@ -3371,7 +3371,7 @@ definitions:
         description: Installer version.
       installation_disk_path:
         type: string
-        description: Host installation path.
+        description: Contains the inventory disk id to install on. This field should actually be called installationDiskID, the field has been repurposed.
```
Comment (slaviered) [inline swagger.yaml:3374]: Too wordy and unclear. Just say 'The ID of the selected installation disk'

File: swagger.yaml [hunk 18]
```diff
@@ -4282,8 +4285,10 @@ definitions:
         type: string
       by_path:
         type: string
+        description: by-path is the shortest physical path to the device
       by_id:
         type: string
+        description: by-id is the wwn/enve-ei which guaranteed to be unique for every storage device
```
Comment (slaviered) [inline swagger.yaml:4291]: World Wide Number of the device

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (rollandf) [reply internal/bminventory/inventory.go]: It is not in swagger because it is not exposed in the API.
If you look at the Cluster, you can see that there also additional fields that are not in swagger.
The field is in the DB, look in #1183 for how it was added.

Comment (rollandf) [reply internal/bminventory/inventory.go]:
```
	return db.AutoMigrate(&models.Host{}, &Cluster{}, &Event{}).Error
	return db.AutoMigrate(&Host{}, &Cluster{}, &Event{}).Error
```

**Comment by slaviered** [comment]
use more concise PR summary: MGMT-4049 Use ID to identify disks instead of disk path

File: internal/host/hostutil/host_utils.go (hunk 10 above)
Comment (omertuc) [reply internal/host/hostutil/host_utils.go:79]: > Agreed. But having migration script to change the content to installation_disk_path can resolve this issue without need for convoluted code in runtime

I'm not sure that's really an option -there are 3 service pods running in production:

`Old`,`Old`,`Old`

In the new version you will migrate (during the service startup I presume) the entire database to use the ID as the installation path rather than path as installation path.

K8s will begin rolling out the new service:

`New`,`Old`,`Old`

While 2 `Old` instances are running, `New` will begin changing the entire database during its startup, while `Old` and `Old are unaware of this change. `Old` and `Old` are still serving requests and they might serve an `install` request for a host already changed by `New` and this might cause problems.

Another issue that might happen is that `Old` and `Old` create new clusters / hosts while `New` is performing the migration - and `New` will miss those and they won't get migrated.

Comment (slaviered) [reply internal/host/hostutil/host_utils.go:79]: Than the best option is to add a new column. It will also solve the compatibility problem of old agents connecting to the new service. Both columns should be maintained in the transition period and than we shall phase out the disk_installation_path

File: swagger.yaml (hunk 17 above)
Comment (omertuc) [reply swagger.yaml:3374]: I requested this description - the field is called "path" but we changed its purpose to "id" and kept the old name.

I think this warrants a clear explanation because it's not intuitive for anyone not familiar with what happened.

A disk has a `path`, `by-path`, `by-id` and `id` (which is confusing enough), a field called installation_disk_path all of a sudden contains the disk ID rather than the path - this adds even more confusion. This explanation clears it up somewhat.

Also the description was meant to be verbose, I don't see a reason saving on words here

File: internal/host/hostcommands/install_cmd.go [hunk 19]
```diff
@@ -235,7 +235,7 @@ func getBootDevice(log logrus.FieldLogger, hwValidator hardware.Validator, host
 
 		return "", newErr
 	}
-	return hostutil.GetDeviceFullName(disks[0].Name), nil
+	return disks[0].ID, nil
```
Comment (omertuc) [inline internal/host/hostcommands/install_cmd.go:238]: The agent expects to receive a path, not an ID

**Comment by slaviered** [comment]
Add a hook to the inventory save() that when the ID is not present (old agents) you are computing it and add it to the structure. This will help UI to handle backward comparability with the disk selection

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go]: look like i can't use this utility method... 
may i resolve this conversation?

File: internal/bminventory/inventory.go (hunk 7 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go]: done

File: internal/bminventory/inventory.go (hunk 8 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go:2166]: The original code set it to an empty string and call to the UpdateInstallationDiskPath method which returns an error. I preserve this behavior.
It possible to change it (make sense), but i'm not sure if to do that as part of this PR.

File: internal/host/host.go (hunk 9 above)
Comment (michaellevy101) [reply internal/host/host.go:262]: @slaviered please see https://issues.redhat.com/browse/MGMT-4049

File: internal/bminventory/inventory.go (hunk 7 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go]: done

File: internal/host/hostutil/host_utils.go (hunk 10 above)
Comment (michaellevy101) [reply internal/host/hostutil/host_utils.go]: please see https://issues.redhat.com/browse/MGMT-4049

File: internal/host/transition.go (hunk 12 above)
Comment (michaellevy101) [reply internal/host/transition.go]: done

File: internal/host/host_test.go (hunk 14 above)
Comment (michaellevy101) [reply internal/host/host_test.go:1072]: done. I didn't add it where the first disk should be selected.

File: internal/host/host_test.go (hunk 13 above)
Comment (michaellevy101) [reply internal/host/host_test.go:1055]: It adds more complexity to the test....

Comment (omertuc) [reply internal/host/host_test.go:1055]: why? just delete `Name: diskName` and I think it will work

File: internal/host/hostcommands/install_cmd.go (hunk 15 above)
Comment (michaellevy101) [reply internal/host/hostcommands/install_cmd.go]: @omertuc The disk id is a symlink to the disk path so it should be fine. I will test it soon.
@slaviered can you please provide more details about your comment? which conversion?

File: internal/host/hostcommands/install_cmd.go (hunk 16 above)
Comment (michaellevy101) [reply internal/host/hostcommands/install_cmd.go]: please review the latest code. I changed it already

File: swagger.yaml (hunk 17 above)
Comment (michaellevy101) [reply swagger.yaml:3374]: Please let me know what's the final decision

File: internal/host/hostcommands/install_cmd.go (hunk 19 above)
Comment (michaellevy101) [reply internal/host/hostcommands/install_cmd.go]: The id is a path....

File: swagger.yaml (hunk 18 above)
Comment (michaellevy101) [reply swagger.yaml]: done

File: internal/host/host_test.go (hunk 13 above)
Comment (michaellevy101) [reply internal/host/host_test.go:1043]: The path is built from the Name field so it makes it more complicated.
please see your next comment where you asked me to add the other_disk_id to the inventory while the test works without id so i aligned it

File: internal/host/hostcommands/install_cmd.go (hunk 15 above)
Comment (slaviered) [reply internal/host/hostcommands/install_cmd.go]: set installation_disk_id (now that we have new field) to /dev/sda and see that it changed to the newer ID

File: internal/controller/config/crd/bases/adi.io.my.domain_agents.yaml [hunk 20]
```diff
@@ -268,6 +272,8 @@ spec:
                           type: string
                         wwn:
                           type: string
+                      required:
+                      - id
```
Comment (omertuc) [inline internal/controller/config/crd/bases/adi.io.my.domain_agents.yaml:276]: Making it required turns it into a pointer, make sure the agent is prepared for that

File: internal/host/hostutil/host_utils.go [hunk 21]
```diff
@@ -68,6 +68,96 @@ func ValidateHostname(hostname string) error {
...
+func UnmarshalInventory(inventoryStr string) (*models.Inventory, error) {
+	var inventory models.Inventory
+
+	if err := json.Unmarshal([]byte(inventoryStr), &inventory); err != nil {
+		return nil, err
+	}
+	return &inventory, nil
+}
+
+func GetHostInstallationPath(host *models.Host) string {
```
Comment (omertuc) [inline internal/host/hostutil/host_utils.go:126]: Can we call disk `GetHostInstallationDisk`?

Comment (omertuc) [inline internal/host/hostutil/host_utils.go:126]: I don't like this name but I don't have better suggestions

File: internal/host/transition.go [hunk 22]
```diff
@@ -106,46 +105,42 @@ func (th *transitionHandler) PostRegisterDuringReboot(sw stateswitch.StateSwitch
...
-
 	if installationDisk == nil {
 		return errors.New(fmt.Sprintf("PostRegisterDuringReboot Could not find installation disk %s for host %s",
-			sHost.host.InstallationDiskPath, *sHost.host.ID))
+			hostInstallationPath, host.ID.String()))
 	}
 
-	statusInfo := fmt.Sprintf("Expected the host to boot from disk, but it booted the installation image - please reboot and fix boot order to boot from disk %s",
-		sHost.host.InstallationDiskPath)
+	var messages = make([]string, 0, 4)
```
Comment (omertuc) [inline internal/host/transition.go:132]: `messages := make([]string, 0, 4)`

Comment (omertuc) [reply internal/host/transition.go:132]: Also I would drop the 4, it's an extremely negligible optimization

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (YuviGold) [reply internal/bminventory/inventory.go]: @michaellevy101 I don't see a reason why we need another function. The common module and functions I made were to avoid this kind of confusion.

Comment (michaellevy101) [reply internal/bminventory/inventory.go]: All the code uses the models.Host and not the common.Host and we fetch the host from the DB 5 times (I think).
I just extracted this code to a new method to be more readable but not changed the logic...
I agree that we should use your new method but IMO this PR is not the right place, let's do this in a dedicated PR

Comment (slaviered) [reply internal/bminventory/inventory.go]: I agree with Yuval's comment. If there is a known utility use it and then pass on just the models.Host portion. I can see getHost() is used only once in this specific commit

Comment (michaellevy101) [reply internal/bminventory/inventory.go]: @slaviered  please see the whole file...

Comment (YuviGold) [reply internal/bminventory/inventory.go]: It was decided to open a new task [MGMT-4801](https://issues.redhat.com/browse/MGMT-4801) to make this refactor aligned throughout the server.
